src/main/java/com/zaxxer/nuprocess/osx/KeventArray.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2015 Ben Hamilton
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.zaxxer.nuprocess.osx;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Ben Hamilton
+ */
+final class KeventArray
+{
+   private final int size;
+   private final Pointer pointer;
+   private final List<Kevent> kevents;
+
+   public KeventArray(int size)
+   {
+      this.size = size;
+      long memory = Native.malloc(32 * size);
+      pointer = new Pointer(memory);
+      kevents = new ArrayList<Kevent>(size);
+      for (int i = 0; i < size; i++) {
+         kevents.add(new Kevent(pointer.share(i * 32)));
+      }
+   }
+
+   public void free()
+   {
+      Native.free(Pointer.nativeValue(pointer));
+      kevents.clear();
+   }
+
+   public Pointer getPointer() {
+      return pointer;
+   }
+
+   public Kevent get(int i) {
+      return kevents.get(i);
+   }
+
+  public int size() {
+    return size;
+  }
+}
```

Comment:
Here is what I'm worried about.  I cheated.  I'm not sure that each Kevent is actually 32 bytes.  I think I studied it and decided that it was *less than 32 bytes*.  It could be as small as 16 bytes...

And because I only read one event at a time, there was no issue.  But when passing an array of kevent objects to the native side the size needs to be exact, because only the address of the first kevent is passed.

Back in the day, Kevent *was* [a structure](https://github.com/brettwooldridge/NuProcess/blob/a10d761f16688936338e091df9d42af0684a9b23/src/main/java/org/nuprocess/osx/Kevent.java).  But I think my mapping wasn't quite right, and I would occassionally get memory access violations.